bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/profiles/ModbusGainOffsetProfile.java

```diff
@@ -231,10 +231,9 @@ private Type applyGainOffset(Type state, boolean towardsItem) {
      * When the conversion is towards the handler (towardsItem=false), unit will be ONE
      *
      */
-    @SuppressWarnings("unchecked") // Safe cast since QU = Dimensionless * QU
     private <QU extends Quantity<QU>> QuantityType<QU> applyGainTowardsItem(QuantityType<Dimensionless> qtState,
             QuantityType<QU> gainDelta) {
-        return (QuantityType<QU>) qtState.multiply(gainDelta);
+        return new QuantityType<>(qtState.toBigDecimal().multiply(gainDelta.toBigDecimal()), gainDelta.getUnit());
```

Comment:
Looks to be correct, tests should reveal any issues. 

I am not fond of bypassing quantitytype-arithmetic since we lose many of the correctness guarantees from the uom lib. Having said this , in this particular case the change is isolated and allows to remove the unchecked cast.